### PR TITLE
Add toggles

### PR DESCRIPTION
**Pull request by kevinch**
#### What is the purpose of this pull request?
Add configuration options.

#### Screenshots or example usage
screen shot 2018-06-11 at 12 31 23

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/ConfigToggle.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import PropTypes from 'prop-types'
+
+import Toggle from '@vtex/styleguide/lib/Toggle'
+
+class ConfigToggle extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      checked: props.checked,
```
Comment (klzns) [inline react/components/ConfigToggle.js]: O estado esta sendo controlado aqui e no pai. Deixar o pai controlar o estado do componente.

Misturar controlled e uncontrolled components pode dar bugs.

Ler esse artigo recem publicado do time do React:
https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

File: react/components/ConfigToggle.js [hunk 2]
```diff
@@ -4,27 +4,14 @@ import PropTypes from 'prop-types'
...
+    this.props.onChange({
+      id: this.props.id,
     })
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.checked !== this.state.checked) {
-      this.setState({ checked: nextProps.checked })
-    }
+    this.setState({ checked: nextProps.checked })
```
Comment (klzns) [inline react/components/ConfigToggle.js]: Nao precisa desse estado/linha/lifecycle

**Comment by kevinch** [comment]
Update: 
screen shot 2018-07-02 at 19 30 26

File: react/config.js [hunk 3]
```diff
@@ -27,6 +29,21 @@ class StoreConfig extends React.Component {
     this.props.fetchStoreConfig()
   }
 
+  componentWillReceiveProps(nextProps) {
```
Comment (klzns) [inline react/config.js]: Podia usar o `getDerivedStateFromProps` ja que esse lifecycle vai morrer na proxima versao do React

Comment (klzns) [reply react/config.js]: E ve se nao ta caindo nesses casos:
https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#when-to-use-derived-state

Comment (kevinch) [reply react/config.js]: Issue criada https://github.com/vtex-apps/credit-control/issues/117

**Pull request merged**